### PR TITLE
Add Transaction support

### PR DESCRIPTION
**Pull request by relistan**
This is a proof of concept that should allow using logrus and passing the New Relic transaction as field. That in turn will associate any fields with the transaction and report it to New Relic with `noticeError`. Additionally, this includes a logrus formatter to remove `txn` from the logs and to allow filtering any sensitive keys in the application that we don't want to log.

### CODE REVIEW

File: formatter.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package newrelic_logrus
+
+import (
+	"github.com/sirupsen/logrus"
+)
+
+// A Formatter is a implements the logrus.Formatter interface and allows
```
Comment (relistan) [inline formatter.go]:
```suggestion
// A Formatter implements the logrus.Formatter interface and allows
```

File: hook.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	}
+
+	return fn(txn)
+}
+
+func (n *NewRelicLogrusHook) Fire(entry *logrus.Entry) error {
+	n.didFire = true // for testing only... there's no way to get data out of NR agent
+
+	return n.withTransaction(entry, func(txn newrelic.Transaction) error {
+		for k, v := range entry.Data {
```
Comment (relistan) [inline hook.go:49]: This is cool because it adds all the other `Field`s as metadata on the New Relic error as well.

File: hook.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+			if k!= "txn" {
```
Comment (andrijaa) [inline hook.go]:
```suggestion
			if k != "txn" {
```

**Review by andrijaa — APPROVED** [review]
🚢

File: hook.go (hunk 3 above)
Comment (relistan) [reply hook.go]: Ran `go fmt`, thanks!

**Pull request merged**